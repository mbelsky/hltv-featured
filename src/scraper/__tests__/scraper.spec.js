jest.mock('axios', () => () =>
  Promise.resolve({ data: require('./matches-example') }),
)

jest.mock('@hltvf/monitoring/alerter', () => ({}))
jest.mock('@hltvf/monitoring/logger', () => ({
  logFabric: () => () => ({}),
}))

jest.mock('common/manageMatches', () => ({
  removeOutdatedMatches: () => Promise.resolve(),
  saveFeaturedMatches: jest.fn(() => Promise.resolve()),
}))

const scrap = require('../scraper')

describe('scraper', () => {
  test('integration', async () => {
    const manageMatches = require('common/manageMatches')
    const { saveFeaturedMatches } = manageMatches

    expect(await scrap()).toBeUndefined()
    expect(saveFeaturedMatches.mock.calls.length).toBe(1)
    expect(saveFeaturedMatches.mock.calls[0]).toMatchSnapshot()
  })
})
