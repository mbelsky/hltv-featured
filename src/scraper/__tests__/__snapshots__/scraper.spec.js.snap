// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scraper integration 1`] = `
Array [
  Array [
    Object {
      "event": "Aorus League 2020 #3 Brazil",
      "href": "https://www.hltv.org/matches/2343039/redemption-poa-vs-imperial-aorus-league-2020-3-brazil",
      "id": 2343039,
      "stars": 0,
      "title": "Redemption POA vs Imperial",
      "unixTimestamp": 1595980800000,
    },
    Object {
      "event": "Nine to Five 2 Swiss Round 3 #2",
      "href": "https://www.hltv.org/matches/2342989/nine-to-five-2-swiss-round-3-2-nine-to-five-2",
      "id": 2342989,
      "stars": 0,
      "title": "TBD vs TBD",
      "unixTimestamp": 1596006000000,
    },
    Object {
      "event": "LPL Pro League Season 5",
      "href": "https://www.hltv.org/matches/2342918/rooster-vs-paradox-lpl-pro-league-season-5",
      "id": 2342918,
      "stars": 0,
      "title": "Rooster vs Paradox",
      "unixTimestamp": 1596099600000,
    },
    Object {
      "event": "Gamers Club Masters V",
      "href": "https://www.hltv.org/matches/2342967/w7m-vs-asterius-gamers-club-masters-v",
      "id": 2342967,
      "stars": 0,
      "title": "W7M vs Asterius",
      "unixTimestamp": 1596128400000,
    },
    Object {
      "event": "Gamers Club Masters V",
      "href": "https://www.hltv.org/matches/2342968/keyd-vs-boom-gamers-club-masters-v",
      "id": 2342968,
      "stars": 0,
      "title": "Keyd vs BOOM",
      "unixTimestamp": 1596133800000,
    },
    Object {
      "event": "Gamers Club Masters V",
      "href": "https://www.hltv.org/matches/2342969/red-canids-vs-bravos-gamers-club-masters-v",
      "id": 2342969,
      "stars": 0,
      "title": "RED Canids vs Bravos",
      "unixTimestamp": 1596139200000,
    },
    Object {
      "event": "Gamers Club Masters V",
      "href": "https://www.hltv.org/matches/2342970/isurus-vs-pain-gamers-club-masters-v",
      "id": 2342970,
      "stars": 0,
      "title": "Isurus vs paiN",
      "unixTimestamp": 1596144600000,
    },
    Object {
      "event": "Showmatch CS:GO",
      "href": "https://www.hltv.org/matches/2342965/navi-2010-vs-natus-vincere-showmatch-csgo",
      "id": 2342965,
      "stars": 1,
      "title": "NAVI 2010 vs Natus Vincere",
      "unixTimestamp": 1596286800000,
    },
    Object {
      "event": "CTRL PLAY League",
      "href": "https://www.hltv.org/matches/2343026/gambit-youngsters-vs-navi-junior-ctrl-play-league",
      "id": 2343026,
      "stars": 0,
      "title": "Gambit Youngsters vs NAVI Junior",
      "unixTimestamp": 1596286800000,
    },
    Object {
      "event": "CTRL PLAY League",
      "href": "https://www.hltv.org/matches/2343025/hard-legion-vs-cyber-legacy-ctrl-play-league",
      "id": 2343025,
      "stars": 0,
      "title": "Hard Legion vs Cyber Legacy",
      "unixTimestamp": 1596297600000,
    },
    Object {
      "event": "ESEA MDL Season 33 Europe Finals",
      "href": "https://www.hltv.org/matches/2343017/ago-vs-sprout-esea-mdl-season-33-europe-finals",
      "id": 2343017,
      "stars": 0,
      "title": "AGO vs Sprout",
      "unixTimestamp": 1596826800000,
    },
    Object {
      "event": "ESEA MDL Season 33 Europe Finals",
      "href": "https://www.hltv.org/matches/2343018/galaxy-racer-vs-ago-sprout-loser-esea-mdl-season-33-europe-finals",
      "id": 2343018,
      "stars": 0,
      "title": "Galaxy Racer vs AGO/Sprout loser",
      "unixTimestamp": 1596913200000,
    },
    Object {
      "event": "ESEA MDL Season 33 Europe Finals Grand Final",
      "href": "https://www.hltv.org/matches/2343019/esea-mdl-season-33-europe-finals-grand-final-esea-mdl-season-33-europe-finals",
      "id": 2343019,
      "stars": 0,
      "title": "TBD vs TBD",
      "unixTimestamp": 1596996000000,
    },
    Object {
      "event": "ESL Australia & NZ Championship Season 11",
      "href": "https://www.hltv.org/matches/2342643/ground-zero-vs-paradox-esl-australia-nz-championship-season-11",
      "id": 2342643,
      "stars": 0,
      "title": "Ground Zero vs Paradox",
      "unixTimestamp": 1597134600000,
    },
    Object {
      "event": "ESL Australia & NZ Championship Season 11",
      "href": "https://www.hltv.org/matches/2342520/bantz-vs-truckers-with-attitude-esl-australia-nz-championship-season-11",
      "id": 2342520,
      "stars": 0,
      "title": "Bantz vs TRUCKERS WITH ATTITUDE",
      "unixTimestamp": 1597147200000,
    },
  ],
]
`;

exports[`scraper integration 2`] = `
Array [
  Array [
    "redemption poa",
    "imperial",
    "tbd",
    "rooster",
    "paradox",
    "w7m",
    "asterius",
    "keyd",
    "boom",
    "red canids",
    "bravos",
    "isurus",
    "pain",
    "navi 2010",
    "natus vincere",
    "gambit youngsters",
    "navi junior",
    "hard legion",
    "cyber legacy",
    "ago",
    "sprout",
    "galaxy racer",
    "ago/sprout loser",
    "ground zero",
    "bantz",
    "truckers with attitude",
  ],
]
`;
